sql/mysqld.cc

```diff
@@ -6360,7 +6360,9 @@ void handle_connections_sockets()
                                     &length);
       if (mysql_socket_getfd(new_sock) != INVALID_SOCKET)
         handle_accepted_socket(new_sock, sock);
-      else if (socket_errno != SOCKET_EINTR && socket_errno != SOCKET_EAGAIN)
+      else if (socket_errno == SOCKET_EAGAIN)
```

Comment:
could you change that test to `(socket_errno == SOCKET_EAGAIN || socket_errno == SOCKET_EWOULDBLOCK)` ?
Linux and FreeBSD documentation mention both of the codes for the case where non-blocking server socket does not have connections that can be accepted, while macOS only mentions EWOULDBLOCK

Reply:
Actually, it's just `EWOULDBLOCK` on Linux, FreeBSD and all the others - but on these, `EAGAIN==EWOULDBLOCK`.
Only on Windows, it's documented as `WSAEINPROGRESS` (for blocking sockets) or `WSAEWOULDBLOCK` (for non-blocking sockets). Windows is weird.
See https://github.com/CM4all/libcommon/blob/master/src/net/SocketError.hxx#L90 (and the surrounding code) for all the weird corner cases....
In any case, adding `EWOULDBLOCK` doesn't hurt - it costs nothing, it's redundant and the compiler will optimize it away.

Reply:
Thanks!

Do not worry about Windows, it does not run this code. The Windows specific code had been factored out into handle_connections_win.cc, which has named pipes, Windows events, and asynchronous APIs, and special stuff like AcceptEx,  more than one "poll" loop like that would offer.

Thanks for mentioning EAGAIN and EWOULDBLOCK are be the same, I never actually checked, alas it is not guaranteed to be everywhere,  there is at least a "historic" Unix (AIX), where we try to build, where those constants are different, according to https://stackoverflow.com/a/7003379/547065.